### PR TITLE
Add conditional xfail test for DISTINCT aggregates with NaN

### PR DESCRIPTION
**Pull request by mythrocks**
SPARK-32038 reports a regression in Apache Spark (3.0.0), in
failing to normalize NaN/Zero float values, during DISTINCT
aggregations. This causes a mismatch in results between
Apache Spark 3.0.0 on CPU, and the Rapids Accelerator (which
returns the right results).
[SPARK-32038](https://issues.apache.org/jira/browse/SPARK-32038) was fixed in apache/spark#28876.

The following change introduces a conditional xfail test that passes
on Apache Spark 3.0.1 and 3.1+ (which fixes SPARK-32038),
but produces an expected failure on Spark 3.0.0.

Testing:
1. On Apache Spark 3.0.0:
```
SPARK_HOME=~/workspace/dev/spark/3.0.0 ./run_pyspark_from_build.sh  -k hash -s

src/main/python/hash_aggregate_test.py::test_count_distinct_with_nan_floats[[('a', RepeatSeq), ('b', Float), ('c', Long)]][IGNORE_ORDER, APPROXIMATE_FLOAT] XFAIL [100%]
...
XFAIL src/main/python/hash_aggregate_test.py::test_count_distinct_with_nan_floats[[('a', RepeatSeq), ('b', Float), ('c', Long)]][IGNORE_ORDER, APPROXIMATE_FLOAT]
  [SPARK-32038][SQL] NormalizeFloatingNumbers should also work on distinct aggregate (https://github.com/apache/spark/pull/28876) Fixed in later Apache Spark releases.
```
2. On Apache Spark 3.0.1:
```
SPARK_HOME=~/workspace/dev/spark/3.0.1 ./run_pyspark_from_build.sh  -k hash -s
...
src/main/python/hash_aggregate_test.py::test_count_distinct_with_nan_floats[[('a', RepeatSeq), ('b', Float), ('c', Long)]][IGNORE_ORDER, APPROXIMATE_FLOAT] PASSED [100%]
```

### CODE REVIEW

**Comment by mythrocks** [comment]
build

**Comment by mythrocks** [comment]
For reference: [xfail Documentation](https://docs.pytest.org/en/latest/skipping.html#xfail-mark-test-functions-as-expected-to-fail)
The initial intention was to introduce an imperative `xfail`, based on the Apache Spark version string. But this would not execute any code after the `xfail` invocation. 
Adding it to the `@pytest.mark.xfail()` lets the test run on Spark 3.0.0, and then excuses the failure. This way, we can look out for a future `xpass`, in case SPARK-32038 is ported back to Spark 3.0.0.

File: integration_tests/src/main/python/hash_aggregate_test.py [hunk 1]
```diff
@@ -338,8 +338,11 @@ def test_hash_agg_with_nan_keys(data_gen):
         conf=_no_nans_float_conf)
 
 
-@pytest.mark.xfail(reason="count(distinct floats) fails when there are NaN values in the aggregation column."
-                          "(https://github.com/NVIDIA/spark-rapids/issues/194)")
+@pytest.mark.xfail(
+    condition=with_spark_session(lambda spark : spark.sparkContext.version == "3.0.0"),
```
Comment (abellina) [inline integration_tests/src/main/python/hash_aggregate_test.py:342]: This is great :+1:

**Comment by mythrocks** [comment]
build

**Comment by mythrocks** [comment]
Hmm... I'm not sure why the status checks are blocked.

**Comment by revans2** [comment]
> Hmm... I'm not sure why the status checks are blocked.

The build for branch-0.2 is not set up yet.  Still waiting on the team to do it.

**Comment by mythrocks** [comment]
> > not sure why the status checks are blocked.
> The build for branch-0.2 is not set up yet. 

D'oh! Thanks, @revans2. Permission to merge this?

**Comment by revans2** [comment]
We have only been merging necessary doc changes. If it is still not up tomorrow then I will start manually testing/merging things.

**Comment by mythrocks** [comment]
> We have only been merging necessary doc changes.

Gotcha. I mistakenly thought test code was acceptable. This patch will keep, so there isn't a rush. Thanks, @revans2.

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Pull request merged**